43771: Show Folder title on Application home pages

labkey-nicka opened the pull request:
#### Rationale
This PR addresses:
* [Issue 43771](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=43771): Show Folder title on Application home pages.
* [Issue 43474](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=43474): Prevent text wrapping for date columns

_43771: Here you can see the folder title in the upper right-hand corner of the page._
![image](https://user-images.githubusercontent.com/3926239/131404704-f57b4983-dd74-4918-8957-ce8451539c51.png)

#### Related Pull Requests
* https://github.com/LabKey/biologics/pull/980
* https://github.com/LabKey/sampleManagement/pull/681
* https://github.com/LabKey/inventory/pull/296

#### Changes
* Pass `QueryColumn` metadata to `<DefaultRenderer/>` to allow for processing of type.
* Add optional `container` and `showFolderTitle` flags to `<UserDetailHeader/>`.
* Replace `detail-display` with specific white-space declaration classes `ws-pre-wrap` and `ws-no-wrap`.

labkey-nicka committed:
Issue 43771: Display container title in UserDetailHeader

labkey-nicka committed:
Issue 43474: Prevent text wrapping for date columns

labkey-nicka committed:
Prepare release notes

labkey-susanh commented on packages/components/src/internal/components/user/UserDetailHeader.spec.tsx:
```diff
@@ -10,40 +10,21 @@ import { UserDetailHeader } from './UserDetailHeader';
-                dateFormat={undefined}
-            />
-        );
+        const component = <UserDetailHeader dateFormat="YYYY-MM-DD" title="Title" user={TEST_USER_READER} />;
```
Are we intentionally not testing the `dateFormat` is undefined?

labkey-nicka replied:
Yes. The `dateFormat` property was marked as required by the interface. I've marked it as optional on this and the underlying `getUserLastLogin` since it is apparently intended to support it. Also, updated the test to pass in no value for `dateFormat` to ensure it is optional.

labkey-susanh commented on packages/components/src/internal/components/user/UserDetailHeader.tsx:
```diff
@@ -1,40 +1,45 @@
+                    &nbsp;{container.title}
+                </div>
+            )}
+            {lastLogin && <div className="detail__header--desc pull-right">Last Login: {lastLogin}</div>}
```
This is not a difference introduced here, but I notice that Biologics leaves off the `Last login` info while LKSM and FM include it.  Any reason we don't want to be consistent here?  I have no strong opinion one way or the other really.

labkey-nicka replied:
I agree it was an inconsistency. Namely, didn't have an easy way for LKB to utilize the `UserProvider` which provides `userProperties` (as you know). Added `useUserProperties` hook and updated app home pages to utilize it. See fdc4b028.

labkey-susanh replied:
Nice. Thanks.

labkey-nicka committed:
Add useUserProperties hook
- deprecate UserProvider, convert userProperties to type Record<string, any>
- UserDetailHeader: mark dateFormat as optional

labkey-nicka committed:
v2.69.0-fb-nick-fix-219.0

labkey-susanh reviewed (approved):
Looks good.

labkey-nicka committed:
lint

labkey-nicka committed:
v2.69.1

labkey-nicka committed:
Update release notes
